packages/feature-toggles/src/create-feature-updater.ts

```diff
@@ -0,0 +1,110 @@
+import { LDClient, LDFeatureStore } from 'launchdarkly-node-server-sdk'
+import {
+    initialiseClient,
+    allToggles,
+    getLaunchDarklyClientWithRetry,
+} from './launch-darkly/client'
+import { FeatureStore } from './launch-darkly/feature-store'
+import {
+    writeFeatureFile,
+    featureStateFileExists,
+    featureFilePath,
+    readFeatureFile,
+    FeatureUpdater,
+} from '.'
+import { FeatureState } from './universal'
+import { Logger } from 'typescript-log'
+
+export interface FeatureUpdaterConfig {
+    featureStateFile: string
+    launchDarklySdkKey?: string
+}
+
+/** Initialises feature toggles in the master */
+export const createFeatureUpdater = async (
+    logger: Logger,
+    config: FeatureUpdaterConfig,
+): Promise<FeatureUpdater> => {
+    const featureStore = new FeatureStore()
+    let ldClient: LDClient | undefined
+    let initialFeatureState: FeatureState | undefined
+
+    try {
+        ldClient = await initialiseClient(
+            config.launchDarklySdkKey,
+            logger,
+            featureStore as LDFeatureStore,
+        )
+
+        if (ldClient) {
+            initialFeatureState = await allToggles(ldClient, logger)
+        }
+    } catch (err) {
+        logger.warn(
+            { err },
+            'Error initialising launch darkly client, falling back to feature file',
+        )
+    }
+
+    if (initialFeatureState) {
+        await writeFeatureFile(config.featureStateFile, initialFeatureState, logger)
+    }
+
+    // We want to read the state from the file to ensure it works.
+    if (!(await featureStateFileExists(config.featureStateFile))) {
```

Comment:
Personal preference would be for the result of the function call to be a variable outside the if statement

Reply:
Good call, fixed